### PR TITLE
Adds  basic implementation of solution tester worker processes

### PR DESCRIPTION
**Pull request by hahnbi**
#### What's this PR do?
- Adds worker processes for queueing up submitted solutions in Redis
#### What are the important parts of the code?
- \server\public\addProblemSolution.html - form to add toy problems or solutions to database
- \worker - code to create worker process to solve solution
#### How should this be tested by the reviewer?
- start redis, postgres, node server.js, node worker\solutionTester.js
- use http://localhost/addProblemSolution.html to add a toy problem and test cases (e.g. nthFibonacci)
- use the same website to add test solution code
- successful solutions should see "victory!" logged in the solutionTester.js terminal
#### What are the relevant tickets? (Please add `closes`, `refs`, etc)

ref #28

### CODE REVIEW

File: server/solutions/solutionController.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 var db = require('../helpers/dbConfig');
 var Solution = require('./solutionModel.js');
+var redis = require('redis');
```
Comment (kweng2) [inline server/solutions/solutionController.js]: Remove this line

File: server/solutions/solutionController.js [hunk 2]
```diff
@@ -20,7 +22,18 @@ module.exports = {
     });
   },
 
-  // POST /api/solutions/:solutionId
+  // POST /api/solutions/:challengeId
+  testSolution: function (req, res, redisClient) {
+    var solutionAttr = {
+      content: JSON.parse(req.body.content),
```
Comment (kweng2) [inline server/solutions/solutionController.js]: Follow convention: 
{soln_str: str, user_handle: str,
socket_id: str}
Also add socket_id

File: worker/solutionTester.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    // var results = JSON.parse(results);
+    var solutionInfo = JSON.parse(results[1]);
+    var challenge = Challenge.forge({
+      id: solutionInfo.challenge_id
+    });
+
+    var output = challenge.fetch()
+    .then(function (challenge) {      
+      var setIsValidTrue = "isValid = true;"
```
Comment (kweng2) [inline worker/solutionTester.js]: Remember to remove this line

File: worker/solutionTester.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      var context = new vm.createContext(sandbox);
+
+      var solutionText = solutionInfo.content;
+      var solutionScript = new vm.Script(solutionText);
+
+      // var testText = JSON.parse(challenge.get('test_suite'));
+      var testText = challenge.get('test_suite');
+      var testScript = new vm.Script(testText);
+
+      var validateScript = setIsValidTrue;
```
Comment (kweng2) [inline worker/solutionTester.js]: Remove this line

File: worker/solutionTester.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    });
+
+    var output = challenge.fetch()
+    .then(function (challenge) {      
+      var setIsValidTrue = "isValid = true;"
+
+      var sandbox = {
+        // chai: require('chai'),
+        assert: require('chai').assert,
+        isValid: false
```
Comment (kweng2) [inline worker/solutionTester.js]: Remove as well.

**Comment by hahnbi** [comment]
Fixed code according to comments by @kweng2

**Pull request merged**